### PR TITLE
AKS/EKS using Longhorn repo Manifests: Deploy MinIO/NFS for Backup Store Creation

### PR DESCRIPTION
**Pull request by roger-ryao**
#### Which issue(s) this PR fixes:
longhorn/longhorn#9955

#### What this PR does / why we need it:
This is to address the issues affecting the AKS and EKS Jenkins jobs after the merge of [39e61f5](https://github.com/longhorn/longhorn-tests/commit/39e61f5731269429a3c680ce9bc1a437096d1b98).

#### Special notes for your reviewer:
@yangchiu 

#### Additional documentation or context
https://github.com/longhorn/longhorn-tests/pull/2173

## Summary by CodeRabbit

- **New Features**
	- Introduced a new function for installing backup stores from the Longhorn repository, enhancing compatibility with different Kubernetes distributions.
- **Bug Fixes**
	- Improved conditional logic for backup store installation based on Kubernetes distribution, ensuring the correct function is called.
	- Corrected a typo in the backup store function name for consistency and clarity.

### CODE REVIEW

File: test_framework/scripts/longhorn-setup.sh [hunk 1]
```diff
@@ -344,6 +344,18 @@ install_backupstores(){
...
+  CIFS_BACKUPSTORE_URL="https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/backupstores/cifs-backupstore.yaml"
+  AZURITE_BACKUPSTORE_URL="https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/backupstores/azurite-backupstore.yaml"
+  kubectl create -f ${MINIO_BACKUPSTORE_URL} \
+                 -f ${NFS_BACKUPSTORE_URL} \
+                 -f ${CIFS_BACKUPSTORE_URL} \
+                 -f ${AZURITE_BACKUPSTORE_URL}
+  setup_azuitize_backup_store
+}
+
 setup_azuitize_backup_store(){
```
Comment (yangchiu) [inline test_framework/scripts/longhorn-setup.sh:359]: The spelling of `azurite` is still incorrect after https://github.com/longhorn/longhorn-tests/pull/2155#issuecomment-2459186164. Could you help scan the codebase again and fix them?

**Review by yangchiu — CHANGES_REQUESTED** [review]
What about `GKE`? Does `GKE` support deploy backup stores on the control plane?

**Comment by roger-ryao** [comment]
> What about `GKE`? Does `GKE` support deploy backup stores on the control plane?

`GKE` has its own independent `longhorn-setup.sh`, so I didn't update it initially. 
You can refer the script here:  https://github.com/longhorn/longhorn-tests/blob/master/pipelines/gke/scripts/longhorn-setup.sh#L141-L146

File: test_framework/scripts/longhorn-setup.sh (hunk 1 above)
Comment (roger-ryao) [reply test_framework/scripts/longhorn-setup.sh]: Update

**Review by yangchiu — APPROVED** [review]
LGTM

**Pull request merged**